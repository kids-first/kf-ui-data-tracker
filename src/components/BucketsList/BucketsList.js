import React from 'react';
import TimeAgo from 'react-timeago';
import {Header, Icon, List} from 'semantic-ui-react';

const BucketsList = ({buckets}) => (
  <List divided>
    {[...buckets]
      .sort((n1, n2) => {
        if (n1.node.deleted && !n2.node.deleted) {
          return 1;
        } else if (!n1.node.deleted && n2.node.deleted) {
          return -1;
        } else {
          if (!n1.node.study && n2.node.study) {
            return 1;
          } else if (n1.node.study && !n2.node.study) {
            return -1;
          } else {
            if (n1.node.study && n1.node.study) {
              return n1.node.study.kfId.localeCompare(n2.node.study.kfId);
            } else {
              return n1.node.name.localeCompare(n2.node.name);
            }
          }
        }
      })
      .map(({node}) => (
        <List.Item disabled={node.deleted} key={node.name}>
          <List.Content>
            {node.study && (
              <List.Content floated="right">
                <List.Header as={Header} size="tiny" textAlign="right">
                  {node.study.kfId}
                </List.Header>
                <List.Description>{node.study.name}</List.Description>
              </List.Content>
            )}
            <List.Header>{node.name}</List.Header>
            <List.Description>
              <List horizontal bulleted>
                <List.Item>
                  Created <TimeAgo date={node.createdOn} live={false} />
                </List.Item>
                {node.deleted && (
                  <List.Item>
                    <Icon name="warning" fitted color="red" /> Deleted
                  </List.Item>
                )}
              </List>
            </List.Description>
          </List.Content>
        </List.Item>
      ))}
  </List>
);

export default BucketsList;
