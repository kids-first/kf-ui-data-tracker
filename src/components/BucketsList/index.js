export {default as BucketsList} from './BucketsList';
