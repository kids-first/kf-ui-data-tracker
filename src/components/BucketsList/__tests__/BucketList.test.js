import React from 'react';
import {render} from '@testing-library/react';
import BucketsList from '../BucketsList';

const buckets = [
  {
    node: {
      id: 'QnVja2V0Tm9kZTpoYXJuZXNzIHN0aWNreSB0ZWNobm9sb2dpZXN=',
      name: 'harness sticky technologies',
      deleted: false,
      createdOn: '2020-02-26T02:29:09+00:00',
      study: null,
    },
  },
  {
    node: {
      id: 'QnVja2V0Tm9kZTpoYXJuZXNzIHN0aWNreSB0ZWNobm9sb2dpZXN=',
      name: 'harness',
      deleted: false,
      createdOn: '2020-02-26T02:29:09+00:00',
      study: {
        id: 'U3R1ZHlOb2RlOlNEX1lXT0RMSjBP',
        name: 'facilitate open-source users',
        shortName: null,
        kfId: 'SD_YWODLJ00',
        createdAt: '2018-04-23T17:46:55+00:00',
        modifiedAt: '2020-03-04T16:11:18.238627+00:00',
      },
    },
  },
  {
    node: {
      id: 'QnVja2V0Tm9kZTpoYXJuZXNzIHN0aWNreSB0ZWNobm9sb2dpZXM=',
      name: 'harness sticky technologies',
      deleted: false,
      createdOn: '2020-02-26T02:29:09+00:00',
      study: {
        id: 'U3R1ZHlOb2RlOlNEX1lXT0RMSjBP',
        name: 'facilitate open-source users',
        shortName: null,
        kfId: 'SD_YWODLJ0O',
        createdAt: '2018-04-23T17:46:55+00:00',
        modifiedAt: '2020-03-04T16:11:18.238627+00:00',
      },
    },
  },
  {
    node: {
      id: 'QnVja2V0Tm9kZTpzeW5kaWNhdGUgZG90LWNvbSBjb250ZW50',
      name: 'syndicate dot-com content',
      deleted: true,
      createdOn: '2019-11-28T13:07:34+00:00',
      study: {
        id: 'U3R1ZHlOb2RlOlNEX1pWMlgxMzRW',
        name: 'optimize 24/7 bandwidth',
        shortName: null,
        kfId: 'SD_ZV2X134V',
        createdAt: '2018-07-01T20:02:17+00:00',
        modifiedAt: '2020-03-04T16:11:18.247815+00:00',
      },
    },
  },
  {
    node: {
      id: 'QnVja2V0Tm9kZTppdGVyYXRlIGltcGFjdGZ1bCBpbmZvLW1lZGlhcmllcw==',
      name: 'iterate impactful info-mediaries',
      deleted: false,
      createdOn: '2019-10-09T17:07:31+00:00',
      study: null,
    },
  },
];

it('renders empty bucket listing', async () => {
  const tree = render(<BucketsList buckets={[]} />);
  expect(tree.container).toMatchSnapshot();
});

it('renders bucket listing', async () => {
  const tree = render(<BucketsList buckets={buckets} />);
  expect(tree.container).toMatchSnapshot();
});
