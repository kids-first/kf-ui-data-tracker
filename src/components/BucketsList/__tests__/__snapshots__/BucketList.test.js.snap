// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bucket listing 1`] = `
<div>
  <div
    class="ui divided list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <div
            class="ui tiny right aligned header header"
          >
            SD_YWODLJ00
          </div>
          <div
            class="description"
          >
            facilitate open-source users
          </div>
        </div>
        <div
          class="header"
        >
          harness
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Created 
              <time
                datetime="2020-02-26T02:29:09.000Z"
                title="2020-02-26T02:29:09+00:00"
              >
                10 months from now
              </time>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <div
            class="ui tiny right aligned header header"
          >
            SD_YWODLJ0O
          </div>
          <div
            class="description"
          >
            facilitate open-source users
          </div>
        </div>
        <div
          class="header"
        >
          harness sticky technologies
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Created 
              <time
                datetime="2020-02-26T02:29:09.000Z"
                title="2020-02-26T02:29:09+00:00"
              >
                10 months from now
              </time>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          harness sticky technologies
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Created 
              <time
                datetime="2020-02-26T02:29:09.000Z"
                title="2020-02-26T02:29:09+00:00"
              >
                10 months from now
              </time>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          iterate impactful info-mediaries
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Created 
              <time
                datetime="2019-10-09T17:07:31.000Z"
                title="2019-10-09T17:07:31+00:00"
              >
                6 months from now
              </time>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="disabled item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="right floated content"
        >
          <div
            class="ui tiny right aligned header header"
          >
            SD_ZV2X134V
          </div>
          <div
            class="description"
          >
            optimize 24/7 bandwidth
          </div>
        </div>
        <div
          class="header"
        >
          syndicate dot-com content
        </div>
        <div
          class="description"
        >
          <div
            class="ui bulleted horizontal list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              Created 
              <time
                datetime="2019-11-28T13:07:34.000Z"
                title="2019-11-28T13:07:34+00:00"
              >
                7 months from now
              </time>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="red warning fitted icon"
              />
               Deleted
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders empty bucket listing 1`] = `
<div>
  <div
    class="ui divided list"
    role="list"
  />
</div>
`;
